### DISCUSSION TITLE
How to fork step process and make new process running within the same enviroment?

### DISCUSSION THREAD

**Discussion by nikolay-bocharov** [community/community#26217] [Actions]
<p>Hi,</p>
<p>I’m running terraform from my Github Actions workflow. One of my Terraform resources uses a ‘post-create’ step to run a bash command.</p>
<pre><code>resource "google\_endpoints\_service" "openapi\_service" {
service\_name = replace(var.espService.url, "https://", "")
project = var.project\_id  
 openapi\_config = var.config  
 provisioner "local-exec" {  
 command = "gcloud beta run services update ${var.espService.name} --set-env-vars ENDPOINTS\_SERVICE\_NAME=${self.service\_name} --project ${var.project\_id} --platform=managed --region=${var.region}"  
 interpreter = ["/bin/bash", "-c"]  
 }  
}
</code></pre>
<p>So this invokes the following bash command </p>
<pre><code>cloud beta run services update ${var.espService.name} --set-env-vars ENDPOINTS_SERVICE_NAME=${self.service_name} --project ${var.project_id} --platform=managed --region=${var.region}
</code></pre>
<p>Please note that the command is initiated by Terraform process (job step) and not by the Github Action runner  (from workflow yaml).</p>
<p>It turns out that my command gets started within a completely different enviroment, perfectly ‘clean’. I don’t have access to previously installed Google SDK (gcloud). Though if run same command from my workflow yaml it works just fine.</p>
<pre><code>- name Run command
 run: gcloud beta run services update $ESP_NAME --set-env-vars ENDPOINTS_SERVICE_NAME=$ENDPOINT --project $PROJECT --platform=managed
</code></pre>
<p>Is there a way to run bash command <strong>from inside the Job Step</strong> process so that it runs within the same enviroment and have access to the same ‘resources’ as command started with <em>run: …</em> from workflow yaml ?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Do you use this <a href="https://github.com/hashicorp/terraform-github-actions" rel="nofollow noopener">Terraform GitHub Actions</a>? As this is a Docker Container action, if you run post-create bash command inside this step, it is executed on docker container not on the github actions runner machine. So when you run scripts in next steps, the environment will change, it will use runner machine environment , not the Terraform docker container.  <br>
If this doesn’t match your scenario, could you please<a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/managing-a-workflow-run#enabling-step-debug-logging" rel="nofollow noopener">enable step debug logging</a> and share your workflow yaml and workflow run logs here? </p>

**Comment by nikolay-bocharov** [comment]
<p>Hi,</p>
<p>Yes I use <a href="https://github.com/hashicorp/terraform-github-actions" rel="nofollow noopener">Terraform GitHub Actions</a>.</p>
<p>Hm … do you know if there is any alternative to <a href="https://github.com/hashicorp/terraform-github-actions" rel="nofollow noopener">Terraform GitHub Actions</a> which is not a docker based? this is quite an annoying limitattion as having access to local-exec would be userful.</p>
<p>Thank you for the reply!</p>

**Comment by Yanjingzhu** [comment]
<p>Sorry to tell you that I don’t know an alternative to that docker container Terraform action. What’s the insufficient of</p>
<p>running  <strong>gcloud </strong> command from your workflow yaml ? </p>

**Comment by nikolay-bocharov** [comment]
<p>yeah that’s what I finally went for! </p>
<p>Thank you.</p>
